File: app/src/main/java/com/github/yongjhih/dagger2/sample/AppModule.java
package com.github.yongjhih.dagger2.sample;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    Application mApplication;

    public AppModule(Application application) {
        this.mApplication = application;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return mApplication;
    }
}

File: app/src/main/java/com/github/yongjhih/dagger2/sample/AppModule.kt
package com.github.yongjhih.dagger2.sample

import android.app.Application

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class AppModule(internal var mApplication: Application) {

    @Provides
    @Singleton
    internal fun provideApplication(): Application {
        return mApplication
    }
}
